app/src/main/java/com/sunasterisk/musixmatch/data/source/local/PlaylistLocalDataSource.java

```diff
@@ -0,0 +1,203 @@
+package com.sunasterisk.musixmatch.data.source.local;
+
+import android.content.ContentResolver;
+import android.content.ContentUris;
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.provider.MediaStore;
+
+import com.sunasterisk.musixmatch.data.model.Playlist;
+import com.sunasterisk.musixmatch.data.model.Track;
+import com.sunasterisk.musixmatch.data.source.Callback;
+import com.sunasterisk.musixmatch.data.source.PlaylistDataSource;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * Created by superme198 on 14,April,2019
+ * in Music_Match__1.
+ */
+public class PlaylistLocalDataSource implements PlaylistDataSource.Local {
+
+    private static final String CONTENT_URI_NAME = "external";
+    private static final String CONTENT_URI_MEDIA = "content://media";
+    private static final int INCREMENT_NUMBER = 1;
+    private static final int FIRST_ELEMENT_INDEX = 0;
+    private static final int ERROR_INDEX = -1;
+    private static final int SIZE = 1;
+    private static PlaylistLocalDataSource sInstance;
+    private ContentResolver mContentResolver;
+
+    private PlaylistLocalDataSource(Context context) {
+        mContentResolver = context.getContentResolver();
+    }
+
+    public static PlaylistLocalDataSource getsInstance(Context context) {
+        if (sInstance == null) {
+            sInstance = new PlaylistLocalDataSource(context);
+        }
+        return sInstance;
+    }
+
+    @Override
+    public void getPlaylist(Callback<List<Playlist>> callback) {
+        AsyncTask<Void, Void, List<Playlist>> asyncTask = new AsyncTask<Void, Void, List<Playlist>>() {
+            @Override
+            protected List<Playlist> doInBackground(Void... voids) {
+                return getLocalPlaylist();
+            }
+
+            @Override
+            protected void onPostExecute(List<Playlist> playlists) {
+                super.onPostExecute(playlists);
+                callback.getDataSuccess(playlists);
+            }
+        };
+        asyncTask.execute();
+    }
+
+    @Override
+    public Uri addSongToPlaylist(long playlistId, Track track) {
+        if (playlistId < 0) {
+            return null;
+        }
+        // Find the greatest PLAY_ORDER in the playlist
+        int base = getLargestPos(playlistId);
```

Comment:
```suggestion
        int lastPlaylistId = getLargestPos(playlistId);
```
đặt ten cho có nghĩa vào em
đặt base  k ai hiểu cả